src/samplers/categorical.jl

```diff
@@ -4,7 +4,7 @@ struct CategoricalDirectSampler <: Sampleable{Univariate,Discrete}
     prob::Vector{Float64}
 
     function CategoricalDirectSampler(p::Vector{Float64})
```

Comment:
this could be made more generic here:
```
function CategoricalDirectSampler(p::V) where {V<:AbstractVector{<:Real}}
```

Reply:
Good point, I think more generally it would be natural to address #743 in this PR